### PR TITLE
I18n: Developer documentation

### PR DESCRIPTION
**Pull request by joshhunt**
**What this PR does / why we need it**:

Adds some initial developer documentation to CONTRIBUTING for how developers can help with getting Grafana translatable :)

### CODE REVIEW

**Review by zsofiakm — APPROVED** [review]

File: contribute/localisation.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+## How to add a new translation phrase
+
+1. Use one of `@lingui/macro`'s React components with the `id`, ensuring it conforms to the guidelines below, with the default english translation. e.g.
+
+```jsx
+import { Trans } from @lingui/macro
+
+const SearchTitle = ({term}) => (
+  <Trans id="search-page.results-title">
```
Comment (axelavargas) [inline contribute/localisation.md:19]: 🤔 Is it possible to add what will be the guidelines for naming the `id`'s?

File: contribute/localisation.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+const ErrorMessage = ({ id, message }) => <Trans id={`errors.${id}`}>There was an error: {message}</Trans>;
+```
+
+2. Upon reload, the default English phrase will appear on the page.
+
+3. Before submitting your PR, run the `yarn i18n:extract` command to extract the messages you added into the `messages.po` file and make them available for translation.
+
+## How translations work in Grafana
+
+Grafana uses the [LinguiJS](https://github.com/lingui/js-lingui) framework for managing translating phrases in the Grafana frontend. We use Lingui to mark up phrases for extraction, to automate extracting phrases into message catalogs for use in translation management systems, and then to load the translated phrases into the frontend for the user's selected locale.
```
Comment (axelavargas) [inline contribute/localisation.md:39]: 🖌️ I think in this section would be awesome if we get a diagram that describes this flow  :)

**Review by axelavargas — COMMENTED** [review]
@joshhunt 👏🏾 this is a great start! I loved the structure you used, and I left some feedback as well.

File: contribute/localisation.md (hunk 2 above)
Comment (axelavargas) [reply contribute/localisation.md:39]: I used this in the past :) https://mermaid-js.github.io/mermaid/#/

**Review by axelavargas — APPROVED** [review]
👏🏾 really nice work @joshhunt. 🚀

**Comment by achatterjee-grafana** [comment]
@osg-grafana - FYI

**Comment by osg-grafana** [comment]
cc @grafana/docs-squad

**Pull request merged**